src/colorizer/CanvasOverlay.ts

```diff
@@ -266,23 +266,22 @@ export default class CanvasOverlay implements IRenderCanvas {
    * Returns true if the inner canvas type does not match the dataset type.
    */
   private doesCanvasTypeNeedUpdate(dataset: Dataset): boolean {
-    // Note: If a dataset has both 2D and 3D frames, this won't trigger a canvas change.
     return (
-      (this.innerCanvasType === CanvasType.CANVAS_2D && !dataset.has2dFrames()) ||
+      (this.innerCanvasType === CanvasType.CANVAS_2D && dataset.has3dFrames()) ||
       (this.innerCanvasType === CanvasType.CANVAS_3D && !dataset.has3dFrames())
```

Comment:
this is confusing it me. without knowing what is going on, I would expect one of these to check for 2dframes, ie canvas type is 3D and has 2D frames?

Reply:
It's possible for a dataset to have both 2D and 3D frames, but we don't currently have a way to toggle between them manually from the UI. For now, I've changed this so any dataset that has 3D frames will cause TFE to preferentially show the 3D canvas.

I'll add a comment/TODO on this!